app/models/user.rb

```diff
@@ -23,8 +24,9 @@ def has_staff?
     self.roles.exists?(user_role: 'staff')
   end
 
-  def assign_default_customer 
-    self.permissions.build(role_id: Role.where(user_role: 'customer').first.id)
+  def assign_default_customer
+    customer_role = Role.find_by(user_role: 'customer')
+    self.permissions.build(role_id: customer_role&.id)
```

Comment:
What will happen if customer_role is nil. Will it build permissions with role_id: nil ?